pkg/rpc/server/http.go

```diff
@@ -3,17 +3,83 @@ package server
 import (
 	"fmt"
 	"net/http"
+
+	"github.com/evstack/ev-node/pkg/config"
+	"github.com/evstack/ev-node/pkg/p2p"
+	"github.com/evstack/ev-node/pkg/store"
 )
 
+// BestKnownHeightProvider should return the best-known network height observed by the node
+// (e.g. min(headerSyncHeight, dataSyncHeight) for full nodes, or header height for light nodes).
+type BestKnownHeightProvider func() uint64
+
 // RegisterCustomHTTPEndpoints is the designated place to add new, non-gRPC, plain HTTP handlers.
 // Additional custom HTTP endpoints can be registered on the mux here.
-func RegisterCustomHTTPEndpoints(mux *http.ServeMux) {
+func RegisterCustomHTTPEndpoints(mux *http.ServeMux, s store.Store, pm p2p.P2PRPC, cfg config.Config, bestKnownHeightProvider BestKnownHeightProvider) {
 	mux.HandleFunc("/health/live", func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Type", "text/plain")
 		w.WriteHeader(http.StatusOK)
 		fmt.Fprintln(w, "OK")
 	})
 
+	// Readiness endpoint
+	mux.HandleFunc("/health/ready", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/plain")
+
+		// Peer readiness: non-aggregator nodes should have at least 1 peer
+		if pm != nil && !cfg.Node.Aggregator {
+			peers, err := pm.GetPeers()
+			if err != nil {
+				http.Error(w, "UNREADY: failed to query peers", http.StatusServiceUnavailable)
+				return
+			}
+			if len(peers) == 0 {
+				http.Error(w, "UNREADY: no peers connected", http.StatusServiceUnavailable)
+				return
+			}
+		}
+
+		localHeight, err := s.Height(r.Context())
+		if err != nil {
+			http.Error(w, "UNREADY: state unavailable", http.StatusServiceUnavailable)
+			return
+		}
+
+		// If no blocks yet, consider unready
+		if localHeight == 0 {
+			http.Error(w, "UNREADY: no blocks yet", http.StatusServiceUnavailable)
+			return
+		}
+
+		// Require best-known height to make the readiness decision
+		if bestKnownHeightProvider == nil {
+			http.Error(w, "UNREADY: best-known height unavailable", http.StatusServiceUnavailable)
+			return
+		}
+
+		bestKnownHeight := bestKnownHeightProvider()
+		if bestKnownHeight == 0 {
+			http.Error(w, "UNREADY: best-known height unknown", http.StatusServiceUnavailable)
```

Comment:
we can turn this into a structured output if it makes things easier to work with cc @auricom  WDYT? We can do a contains `UNREADY` / `READY` from a liveness probe check like this.